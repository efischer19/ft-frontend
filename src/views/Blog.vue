<template>
  <div class="post">
    <template v-for="element in postContent">
      <BlogTitle
        v-if="element.type === 'title'"
        v-bind:msg="element.msg"
        :key="element.index"
      />
      <BlogBodyGraph
        v-else-if="element.type === 'p'"
        v-bind:rawMsg="element.rawMsg"
        v-bind:links="element.links"
        :key="element.index"
      />
      <BlogImg
        v-else-if="element.type === 'img'"
        v-bind:baseFileLink="`/${resolvedId}/${element.link}`"
        v-bind:altText="element.alt"
        :key="element.index"
      />
      <BlogHeader
        v-else-if="element.type === 'header'"
        v-bind:msg="element.msg"
        :key="element.index"
      />
      <BlogList
        v-else-if="element.type === 'list'"
        v-bind:items="element.items"
        :key="element.index"
      />
    </template>
  </div>
</template>

<script>
import axios from 'axios';
import Cookie from 'js-cookie';

import BlogTitle from '@/components/BlogTitle.vue';
import BlogBodyGraph from '@/components/BlogBodyGraph.vue';
import BlogImg from '@/components/BlogImg.vue';
import BlogHeader from '@/components/BlogHeader.vue';
import BlogList from '@/components/BlogList.vue';

export default {
  name: 'blog',
  props: ['id'],
  data() {
    let newId = this.id;
    const isPrivatePost = (this.id.charAt(0) === '_');
    if (!this.ajaxData) {
      this.ajaxData = [{ index: 0, type: 'title', msg: 'Error Loading Post Data' }];
      if (isPrivatePost) {
        const authToken = Cookie.get('ft-auth-token');
        if (authToken) {
          // I could change the error message to something like "not signed in"
          // I'm just leaving the generic error for now though

          newId = this.id.substring(1);
          axios.get(
            `https://nccu1znzcj.execute-api.us-east-2.amazonaws.com/Prod/post-data/${newId}`,
            {
              headers: {
                'X-Api-Key': 'oZE5pkcS5H4PlndobzKdH9wmoxO9uLqa2tY5wtaH',
                Authorization: authToken,
              },
            },
          ).then(({ data }) => {
            this.postContent = data;
          });
        }
      } else {
        axios.get(`/api/public/${this.id}/post_data.json`).then(({ data }) => {
          this.postContent = data;
        });
      }
    }
    return {
      postContent: this.ajaxData,
      resolvedId: newId,
    };
  },
  components: {
    BlogTitle,
    BlogBodyGraph,
    BlogImg,
    BlogHeader,
    BlogList,
  },
};
</script>
