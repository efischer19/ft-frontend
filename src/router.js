import Vue from 'vue';
import Router from 'vue-router';
import Home from './views/Home.vue';
import About from './views/About.vue';
import Blog from './views/Blog.vue';

Vue.use(Router);

// just for local development
var localPostData = require('./mc_data.json');

export default new Router({
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home,
    },
    {
      path: '/about',
      name: 'about',
      component: About,
    },
    {
      path: '/blog',
      name: 'blog',
      component: Blog,
      props: {
        baseImgDir: "mc/",
        postContent: localPostData,
      },
    },
  ],
});
