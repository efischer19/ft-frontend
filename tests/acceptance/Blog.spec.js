import { createRenderer } from 'vue-server-renderer';
import mockAxios from 'jest-mock-axios';
import Vue from 'vue';

import Blog from '@/views/Blog.vue';
import router from '@/router';

const mockResponseObj = require('./post_data.json');

describe('Blog', () => {
  it('should match snapshot', () => {
    const Constructor = Vue.extend(Blog);
    const vm = new Constructor({
      router,
      propsData: {
        id: 'mc',
      },
    }).$mount();

    expect(mockAxios.get).toHaveBeenCalledWith('/api/public/mc/post_data.json', { headers: {} });
    mockAxios.mockResponse({ data: mockResponseObj });

    const renderer = createRenderer();
    Vue.nextTick(() => {
      renderer.renderToString(
        vm,
        (err, str) => {
          if (err) throw new Error(err);
          expect(str).toMatchSnapshot();
        },
      );
    });
  });
});
