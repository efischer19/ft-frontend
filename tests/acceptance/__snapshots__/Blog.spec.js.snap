// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog should match snapshot 1`] = `
<div class="post">
  <h1>Mac And Cheese: Quick and Easy</h1>
  <p>If you are anything like me, you really appreciate a good mac and cheese. I used to make it with a roux-based sauce, which was decent, but not great. It had decent texture, but the flavor was lacking and I always wanted to have something else to serve
    alongside it. Enter
    <a href="https://www.foodnetwork.com/recipes/alton-brown/stovetop-mac-n-cheese-recipe-1939465">Alton Brown</a>. This one's a game-changer, folks. It's so good, and so easy that I've enshrined the recipe on a post-it right in the middle of my kitchen cabinets, where it'll stay until I've commited it to memory thanks to rote repitition.</p>
  <a
    href="/api/public/mc/20180519_mc_postit.jpg">
    <img src="/api/public/mc/20180519_mc_postit.small.jpg" alt="The post-it note I wrote for this recipe, text reproduced below">
    <div class="hidden">Click for high-resolution image</div>
    </a>
    <h2>The Base Recipe</h2>
    <ul>
      <li>1/2 pound noodles, cooked al dente</li>
      <li>4 Tablespoons butter, unsalted</li>
      <li>6 ounces evaporated milk</li>
      <li>2 large eggs</li>
      <li>1 teaspoon salt</li>
      <li>1 teaspoon ground mustard</li>
      <li>10 ounces cheese, shredded</li>
      <li>black pepper, to taste</li>
    </ul>
    <p>Return noodles to their pot, and melt butter into them over low heat. In a measuring cup, mix milk/eggs/salt/mustard together. Pour mixture over butter-coated noodles, and stir together. Gradually add cheese and stir over low heat until it's all melted,
      ~3 minutes. Portion into bowls and top with pepper as desired. Makes 3-4 servings as a side, or 2 as a standalone meal.</p>
    <a href="/api/public/mc/20180519_mc_bowl_shot_1.jpg">
      <img src="/api/public/mc/20180519_mc_bowl_shot_1.small.jpg" alt="2 finished bowls of mac and cheese, ready to eat">
      <div class="hidden">Click for high-resolution image</div>
    </a>
</div>
`;
